addon/mixins/application-route-mixin.js

```diff
@@ -40,6 +38,15 @@ export default Ember.Mixin.create(ApplicationRouteMixin, {
       session.invalidate().then(function () {
         route.transitionTo('login');
       });
+    },
+    documentCreated: function (doc) {
+      this.transitionToDocument(doc);
+    }
+  },
+  
+  transitionToDocument: function (doc) {
+    if (doc) {
+      this.transitionTo('document', doc);
     }
```

Comment:
It's been there in the code I copied it from (in the old `.createFromState().then()`). I guess there used to be a chance nothing would be returned. I can look into it and see if there's still a possibility for this.